admin/javascript/src/LeftAndMain.EditForm.js

```diff
@@ -121,37 +123,37 @@ $.entwine('ss', function($){
 			// full <form> tag by any ajax updates they won't automatically reapply
 			this.changetracker(this.getChangeTrackerOptions());
 		},
-	
+
 		/**
 		 * Function: confirmUnsavedChanges
-		 * 
+		 *
 		 * Checks the jquery.changetracker plugin status for this form,
 		 * and asks the user for confirmation via a browser dialog if changes are detected.
 		 * Doesn't cancel any unload or form removal events, you'll need to implement this based on the return
 		 * value of this message.
-		 * 
+		 *
 		 * If changes are confirmed for discard, the 'changed' flag is reset.
-		 * 
+		 *
 		 * Returns:
-		 *  (Boolean) FALSE if the user wants to abort with changes present, TRUE if no changes are detected 
+		 *  (Boolean) FALSE if the user wants to abort with changes present, TRUE if no changes are detected
 		 *  or the user wants to discard them.
 		 */
 		confirmUnsavedChanges: function() {
 			this.trigger('beforesubmitform');
-			if(!this.is('.changed')) {
+			if(!this.is('.changed') || this.is('.discardchanges')) {
 				return true;
 			}
 			var confirmed = confirm(i18n._t('LeftAndMain.CONFIRMUNSAVED'));
 			if(confirmed) {
 				// confirm discard changes
-				this.removeClass('changed');
+				this.addClass('discardchanges');
```

Comment:
Sure;
The original behaviour was intended to remove the changed flag, but this relied on the class not yet being re-applied during navigation.

The issue is that, perhaps due to some onNavigation hook somewhere, tinymce is doing some kind of "cleanup" that yet again triggers change detection, and re-adds the changed class.

As a result, you ended up with the notification being shown TWICE.

If you say "no I don't want to save my changes" then it doesn't matter if there are changes made immediately after, you have already specified your intention to abandon these changes, so it shouldn't matter if "changed" class is added again.

The solution to this is to "fix" the users decision with a custom add-only class (which can only be removed by re-loading the form).